layout-management-webapps/src/main/webapp/vue-app/site-navigation/components/SiteNavigationAddElementDrawer.vue

```diff
@@ -97,55 +98,93 @@ export default {
 
   data() {
     return {
-      elementType: 'newPage',
-      openMode: 'sameTab',
+      elementType: 'PAGE',
+      openMode: 'SAME_TAB',
       link: '',
       linkRules: [url => !!(url && url.match(/^((https?:\/\/)?(www\.)?[a-zA-Z0-9]+\.[^\s]{2,})|(javascript:)|(\/portal\/)/))
               || ( !url.length && this.$t('siteNavigation.required.error.message') || this.$t('siteNavigation.label.invalidLink'))],
+      navigationNode: null,
+      elementLabel: null,
+      pageTemplate: 'empty',
+      selectedPage: null,
+      loading: false,
     };
   },
   computed: {
     elementTypes() {
       return [
         {
           text: this.$t('siteNavigation.label.newPage'),
-          value: 'newPage',
+          value: 'PAGE',
         },
         {
           text: this.$t('siteNavigation.label.existingPage'),
           value: 'existingPage',
         },
         {
           text: this.$t('siteNavigation.label.link'),
-          value: 'link',
+          value: 'LINK',
         },
       ];
     },
     openModes() {
       return [
         {
           text: this.$t('siteNavigation.label.sameTab'),
-          value: 'sameTab',
+          value: 'SAME_TAB',
         },
         {
           text: this.$t('siteNavigation.label.newTab'),
-          value: 'newTab',
+          value: 'NEW_TAB',
         },
       ];
     },
     isLinkElement() {
-      return this.elementType === 'link';
+      return this.elementType === 'LINK';
     },
   },
   created() {
     this.$root.$on('open-add-element-drawer', this.open);
+    this.$root.$on('page-template-changed', this.changePageTemplate);
+    this.$root.$on('existing-page-selected', this.changeSelectedPage);
+
   },
   methods: {
-    open() {
+    open(elementLabel, navigationNode) {
+      this.elementLabel = elementLabel;
+      this.navigationNode = navigationNode;
       this.$refs.siteNavigationAddElementDrawer.open();
     },
     close() {
       this.$refs.siteNavigationAddElementDrawer.close();
+    },
+    changePageTemplate(pageTemplate) {
+      this.pageTemplate = pageTemplate;
+    },
+    changeSelectedPage(selectedPage) {
+      this.selectedPage = selectedPage;
+    },
+    createElement() {
+      if (this.elementType === 'existingPage') {
+        const pageRef = this.selectedPage?.pageContext?.key?.ref || `${this.selectedPage?.pageContext?.key.site.typeName}::${this.selectedPage?.pageContext?.key.site.name}::${this.selectedPage?.pageContext?.key.name}`;
+        this.$root.$emit('save-node-with-page', {
+          'pageRef': pageRef,
+          'nodeTarget': this.openMode,
+          'pageType': this.elementType
+        });
+      } else {
+        this.$siteNavigationService.createPage(this.elementLabel, this.elementType, this.link, this.pageTemplate, this.navigationNode.siteKey.name, this.navigationNode.siteKey.type)
```

Comment:
normally for the new created page, its name should be the node name not the node label